src/Tasks/GenerateResource.cs

```diff
@@ -2432,6 +2432,12 @@ internal Assembly ResolveAssembly(object sender, ResolveEventArgs args)
                     {
                         if (String.Compare(requestedAssemblyName.Name, candidateAssemblyName.Name, StringComparison.CurrentCultureIgnoreCase) == 0)
                         {
+                            string itemSpec = _assemblyFiles[i].ItemSpec;
+
+                            if (requestedAssemblyName.Name.Contains("System.Runtime"))
```

Comment:
Great question! It "works" with the current code but I have very little confidence in it.

@ericstj I think this is a facade-related question.  I added this code based on some [internal source](http://index/?query=%60because%20the%20parent%20does%20not%20exist&rightProject=Microsoft.VisualStudio.IDE.ToolboxControlsInstaller&file=SharedUtilities.cs&line=393) that resolved the same `"Could not load type 'System.Object' from assembly 'System.Runtime, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' because the parent does not exist."` error that I got when `ReflectionOnlyLoadFrom`ing without it.

I have two questions:
1. In what cases should I load a different assembly, rather than the referenced one?
2. When the load should be redirected, where should it be redirected to?